compiler/qsc/src/interpret.rs

```diff
@@ -567,8 +567,8 @@ impl Interpreter {
         })
     }
 
-    // Invokes the given callable with the given arguments using the current environment and compilation but with a fresh
-    // simulator configured with the given noise, if any.
+    // Invokes the given callable with the given arguments using the current compilation but with a fresh
+    // environment and simulator configured with the given noise, if any.
     pub fn invoke_with_noise(
```

Comment:
Does this change the behavior of invoking `qsharp.code.*` at all? Or is this only used for `.run()`?

Reply:
This path is only used by `qsharp.run` when a callable is passed in. So now we (correctly) only use `self.env` in two places: `eval_increment` which is what cells and `qsharp.eval` sit on top of, and `invoke` which is what actually calling those python wrapper functions uses. Everything else (circuits, running shots, resource estimation, partial eval for QIR) uses fresh environments.